awx/main/management/commands/run_dispatcher.py

```diff
@@ -76,7 +76,7 @@ def handle(self, *arg, **options):
         consumer = None
 
         try:
-            queues = ['tower_broadcast_all', get_local_queuename()]
+            queues = ['tower_broadcast_all', 'tower_settings_change', 'rsyslog_configurer', get_local_queuename()]
```

Comment:
I'm confused about this change (and I don't know dispatcher enough yet - I might just be way off base here).

AFAIK we're not using the dispatcher for either of these daemons (the cache buster nor the rsyslog configurererer). The daemons themselves that are listening for these messages are what is doing the listening for these messages (as opposed to dispatcher), right?

Reply:
I'm not completely clear on all the functions of dispatcher as well but in the discussion on the JIRA ticket it was said that the new rsyslog channel should be added to the queues dispatcher is listening to. I assumed that meant I should add the new cache clear channel as well. Hao has clarified that this will be a service not another container though so I would assume it actually doesn't need to be added.

Reply:
The change here to add "tower_settings_change" is coherent, because it is a `@task`. However, adding the "rsyslog_configurer" would require corresponding changes that are not in this PR. Functionally, a PATCH to `/api/v2/settings/logging/` produces:

```
tools_awx_1 | Traceback (most recent call last):
tools_awx_1 |   File "/awx_devel/manage.py", line 8, in <module>
tools_awx_1 |     manage()
tools_awx_1 |   File "/awx_devel/awx/__init__.py", line 201, in manage
tools_awx_1 |     execute_from_command_line(sys.argv)
tools_awx_1 |   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
tools_awx_1 |     utility.execute()
tools_awx_1 |   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/django/core/management/__init__.py", line 413, in execute
tools_awx_1 |     self.fetch_command(subcommand).run_from_argv(self.argv)
tools_awx_1 |   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/django/core/management/base.py", line 354, in run_from_argv
tools_awx_1 |     self.execute(*args, **cmd_options)
tools_awx_1 |   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/django/core/management/base.py", line 398, in execute
tools_awx_1 |     output = self.handle(*args, **options)
tools_awx_1 |   File "/awx_devel/awx/main/management/commands/run_dispatcher.py", line 81, in handle
tools_awx_1 |     consumer.run()
tools_awx_1 |   File "/awx_devel/awx/main/dispatch/worker/base.py", line 185, in run
tools_awx_1 |     self.process_task(json.loads(e.payload))
tools_awx_1 |   File "/usr/lib64/python3.9/json/__init__.py", line 346, in loads
tools_awx_1 |     return _default_decoder.decode(s)
tools_awx_1 |   File "/usr/lib64/python3.9/json/decoder.py", line 337, in decode
tools_awx_1 |     obj, end = self.raw_decode(s, idx=_w(s, 0).end())
tools_awx_1 |   File "/usr/lib64/python3.9/json/decoder.py", line 355, in raw_decode
tools_awx_1 |     raise JSONDecodeError("Expecting value", s, err.value) from None
tools_awx_1 | json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

This is because traffic in the "rsyslog_configurer" contains non-JSON data, because it is not using a `@task` implementation.

Right now, I would probably advocate removing "rsyslog_configurer" from the queues here and dealing with it in another PR. Either way, the rsyslog re-configuration needs an additional patch because right now the messages are not processed in VM installs.